### DISCUSSION TITLE
Does the repeat function have the usage of two parameters, m and n, which represent the upper and lower limits of the number of repetitions?

### DISCUSSION THREAD

**Discussion by bytesnail** [tree-sitter/tree-sitter#2436] [Q&A] [answered]
What I've learned so far is that we have repeat and repeat1 for any number of repetitions and at least once, respectively, but I don't seem to see how to express the upper and lower limits of arbitrary repetitions.

If I want to match 4 digits, without using regular expressions, it seems that for now I can only write like this:
```
seq($.DIGIT, $.DIGIT, $.DIGIT, $.DIGIT)
```

I'd like to see a simple way to write it like this:
```
repeat($.DIGIT, 4, 4)  or  repeat($.DIGIT, 2, 4)
```

**Comment by ahelwer** [comment, marked as answer]
You can write your own javascript functions implementing whatever you like.
